Locker/Helpers/LockerHelpers.swift

```diff
@@ -107,6 +120,14 @@ extension LockerHelpers {
     static var keyKeychainServiceName: String {
         return "\(LockerHelpers.bundleIdentifier)_KeychainService"
     }
+
+    // MARK: - Device list
+
+    static func fetchNewDeviceList() {
+    #if !targetEnvironment(simulator)
+        devices.fetchDevices()
+    #endif
+    }
```

Comment:
You should check if the current device is already in the list. If it is, then we should not execute an API call for fetching a new (updated) list.